### ISSUE TITLE
Thunderbird crashes: Incompatibility to Master Password+ / StartupMaster

### ISSUE THREAD

**Issue by dodmi** [bstreiff/unread-badge#14]
Hi,

Thunderbird crashes if Unread Badge and an addon to query the master password at startup is used.

I've set up a master password for my profile and am using Master Password+ to query the master password once at startup. I've tried also replacing Master Password+ with StartupMaster (both available add AMO)
In combination with Unread Badge Thunderbird crashes instead of displaying the password prompt.
It would be very nice if you could fix this issue.

My setup:
Windows 10
Thunderbird 52.7.0
Unread Badge 1.2.20180408
Master Password+ 1.29.2 / alternative: StartupMaster 1.6.5

**Comment by dodmi** [comment]
Hi, 
in the meantime I learned many things about Thunderbird addons and had a chance to look at the issue myself.

The good news (for all):
I've identified the problem: These password addons interrupt main window loading.
If the main window is in "this state" and UnreadBadge tries to create the badge (triggered by the "domwindowopened" event), Thunderbird cashes.
Delaying the execution after the window loading completes, solves the problem.
I've fixed this in 1.2.20180408 by replacing
```
if (aTopic == "domwindowopened")
{
       if (!gActiveWindow) setActiveWindow(win);
}
```
with
```
if (aTopic == "domwindowopened")
{
       // run code after window is loaded
       win.addEventListener("load", function listener() {
            win.removeEventListener("load", listener, false);
            if (!gActiveWindow) setActiveWindow(win);
       }, false);            
}
```
in the `observe` function in `bootstrap.js`.

The bad news (for you):
I'm still on Thunderbird 52 because of numerous addons I want to keep.
So you need to help yourself and adapt this for newer addon versions, in case this is still an issue there.

**Comment by dodmi** [comment]
@bstreiff I closed this issue, since it's solved for me. In case your still looking for a solution, feel free to use it. And thank you for the addon.
